tidy3d/plugins/mode/mode_solver.py

```diff
@@ -993,6 +1028,7 @@ def to_mode_solver_monitor(self, name: str, colocate: bool = None) -> ModeSolver
             name=name,
         )
 
+    @require_fdtd_simulation
     def sim_with_source(
```

Comment:
is it possible to modify these to work with EME simulations? in principle they just create a modal source / monitors and update the simulation?

Reply:
Not really, because `EMESimulation` does not currently support any sources, or `ModeMonitor`. For `ModeSolverMonitor`, it is actually supported, so I'll get rid of the requirement there.